scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -188,6 +189,8 @@ class FossId(
     private val waitForResult: Boolean
     private val secretKeys = listOf("serverUrl", "apiKey", "user")
     private val namingProvider: FossIdNamingProvider
+    // a list of all created scans, to be able to delete them in case of error
+    private val createdScans = mutableSetOf<String>()
```

Comment:
Could this be made a local variable of the `scanPackages()` function? I think, the scope of this field is larger than it needs to be.

Reply:
It need to be accessed by `createScan` ... A global variable is better than modifying a parameter no ?

Reply:
But `createScan()` returns the scan code, so the caller could store it in a local variable.

Reply:
Yes but `createScan` is called by `checkAndCreateScan` and `checkAndCreateDeltaScan`.
You could argue that each one of them is returning the `scanCode` to `scanPackages` where the deletion should occur.

However the problem is that in the second branch of `checkAndCreateScan`, no scan is created: previous scan is found and reused. The latter is a scan created in a previous separate run therefore it should not be deleted.

(We delete scans only from this run).

Reply:
Ok, I won't insist.